api/admin/controller.py

```
@@ -383,6 +409,7 @@ def details(self, identifier_type, identifier):
 
         This includes relevant links for editing the book.
         """
+        self.require_librarian(flask.request.library)
```

I'd also like to move in the direction of raising more exceptions that can be converted to problem details rather than returning problem details, I think it makes the code cleaner.

---

What if a ProblemDetail _was_ an exception that could be raised instead of returned?

---

That could work, I might try that but it maybe it would be better to do it in a separate pr.

---

Yeah, that's a big change.